Adding change notes.md

```diff
@@ -0,0 +1,69 @@
+# Adding change notes for query and library changes
```

Comment:
I suggest placing this file in https://github.com/github/codeql/tree/main/docs, and naming it something like `change-notes.md` or `change-notes-guide.md`.

Reply:
Moved to `docs/change-notes.md`, and added a link from `CONTRIBUTING.md`.